### ISSUE TITLE
Ref #11902 - Sample program cause 207 when run from NGINX

### ISSUE THREAD

**Issue by AnnTonyt** [curl/curl#12021]
### I did this

Ref: #11902

Please find sample code with which the curl out of memory issue is recreated when run from NGINX

The https request works fine when the CURL request is sent via the standalone application.
the same code when executed from NGINX gives CURL out of memory error.

curl --version
curl 7.87.0 (powerpc-ibm-aix7.2.4.0) libcurl/7.87.0 OpenSSL/1.1.1l
Release-Date: 2022-12-21
Protocols: dict file ftp ftps gopher gophers http https imap imaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS HSTS HTTPS-proxy IPv6 Largefile NTLM NTLM_WB SSL TLS-SRP UnixSockets

```c
#include
#include "curl.h"

int main() {
CURL* curl;
CURLcode res;

// Initialize libcurl
curl = curl_easy_init();
if (curl) {
    // Set the target URL
    curl_easy_setopt(curl, CURLOPT_URL, "https://host-ip:port#/ndb/routelink/v1/inbound.api/cpr?sha1=123");

    // Set the HTTP method to GET
    curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "GET");

    // Set the 'Content-Type' header
    struct curl_slist* headers = NULL;
    headers = curl_slist_append(headers, "Content-Type: application/json");

    // Set the 'Authorization' header
    headers = curl_slist_append(headers, "Authorization: Bearer JWT_token");

    curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);

    // Allow connections to SSL/TLS sites without certificates
    curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0L);
    curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0L);

	
    // Perform the HTTP request
    res = curl_easy_perform(curl);

    // Check for errors
    if (res != CURLE_OK) {
        fprintf(stderr, "curl_easy_perform() failed: %s\n", curl_easy_strerror(res));
    }

    // Clean up
    curl_easy_cleanup(curl);

    // Free the headers
    curl_slist_free_all(headers);

    // Cleanup libcurl
    curl_global_cleanup();
}

return 0;
}
```

### curl/libcurl version

curl 7.87.0 (powerpc-ibm-aix7.2.4.0) libcurl/7.87.0 OpenSSL/1.1.1l
Release-Date: 2022-12-2

### operating system

AIX

**Comment by dfandrich** [comment]
Can you reproduce this with 8.3.0? There have been a number of instances of
CURLE_OUT_OF_MEMORY errors being replaced with more accurate error codes in the
10 curl releases since the one you're using. And I doubt you're actually
looking for "no response".

As an aside, using CURLOPT_CUSTOMREQUEST is almost never right, including in
this case, and disabling CURLOPT_SSL_VERIFY* can be dangerous.

**Comment by bagder** [comment]
> cause 207 when run from NGINX

What does this mean?

**Comment by jay** [comment]
Add a line `curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L);` so you can see what is happening

[CURLOPT_VERBOSE doc](https://curl.se/libcurl/c/CURLOPT_VERBOSE.html)

FWIW I tried this sample in Windows with url `httpbin.org/get` and the result was CURLE_OK as expected.

**Comment by AnnTonyt** [comment]
The issue is fixed with 8.3.0 version.

Thank you for the support
